sqlair.go

```diff
@@ -0,0 +1,172 @@
+package sqlair
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/canonical/sqlair/internal/expr"
+)
+
+// Statement is a sql statement with valid sqlair DML.
+type Statement struct {
```

Comment:
I think this comment doesn't tell the whole story, perhaps
```
// Statement represents a SQLair DML expression, parsed from a
// SQL query and with relevant type information.
```
Also, is it worth making the capitalisation of mentions of SQL consistent with SQLair?

Reply:
Changed.